core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -231,6 +303,29 @@ class KafkaZkClient(zooKeeperClient: ZooKeeperClient, isSecure: Boolean) extends
     retryRequestUntilConnected(setDataRequest)
   }
 
+  /**
+   * Sets the topic znode with the given assignment.
+   * @param topic the topic whose assignment is being set.
+   * @param assignment the partition to replica mapping to set for the given topic
+   * @throws KeeperException if there is an error while setting assignment
+   */
+  def setTopicAssignment(topic: String, assignment: Map[TopicPartition, Seq[Int]]) = {
+    val setDataResponse = setTopicAssignmentRaw(topic, assignment)
+    if (setDataResponse.resultCode != Code.OK){
+      setDataResponse.resultException.foreach(e => throw e)
```

Comment:
Perhaps we could use the following consistently.

if (code != Code.OK)
   throw KeeperException.create(code)

Reply:
Hmm, that is equivalent to:

```scala
setDataResponse.resultException.foreach(e => throw e)
```

That is, there's no need to check the code and each method doesn't need to decide which exception to throw. We could make it even more concise by adding a `maybeThrow` method (like we have in `Errors`) that would throw the exception if there is one. Then the code would be:

```scala
setDataResponse.maybeThrowResultException()
```

Reply:
Thats good suggestion. I will do this in separate minor PR.